pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -93,6 +96,7 @@ func NewReconciler(mgr manager.Manager, immediateReconcileRequests chan event.Ge
 			ResourceNameLabel: metrics.ResourceNameLabel,
 		},
 		jitterGenerator: jg,
+		oldObjectsCache: &sync.Map{},
```

Comment:
How hard is it to switch to using the informer library? It has the old object caching built in.